Processes.md

```diff
@@ -21,42 +21,114 @@ Up to date
 Maintain state.
 
 
-Let’s create a module Menuprocess.ex and see how we shall execute the process from the client in the menuserver.
+Let’s create a module MenuProcess.ex and see how we shall execute the process from the client in the menuserver.
 Below is a breakdown of a couple of processes in our module. Our module will enable three processes;
 Return the current menu state
 Add items to the menu
 Handle wrong entries
 
-For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own
-process then passed the initial state which in our case will change from an empty [] to an actual list of items to be 
-added to the menu. The function responsible for that is shown below;
+start_link begins a spawn linked to the current process with the given functions, the pid is registered using `Process.register`. For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own process then passed the initial state which in our case will change from an empty [] to an actual list of items to be added to the menu. The function responsible for that is shown below;
 
-**add code**
+ ```
+defmodule Menuprocess do
+  
+  #menu server
+  @doc"""
+  This function initializes our server 
+  """
+   @spec start_link(any()) :: pid()
+    def start_link(initial_state \\ []) do
+     pid = spawn(__MODULE__, :menu_state, [initial_state])
+     Process.register(pid, :process)
+     pid 
+    end
+end 
+```
 
 Return the current menu state:
 To return the current state of the menu, we basically need to send a message from the client requesting for all items
 present in our server to our mailbox which is handled by the receive do (whose work is to await requests and match them 
 to the messages in the mailbox), then the server responds with the state of the menu.
 
-**add code**
+```
+#server
+
+@spec menu_state(tuple()) :: {:ok, list(String.t())}
+  def menu_state(state) do
+    receive do
+      {caller, :items} ->
+        send caller, state
+        menu_state(state)
+    end
+  end
+  
+#from client
+
+@spec all() :: {:ok, list(String.t())}
+    def all do 
+      send :process, {self(), :items}
+        receive do state -> inspect state end
+    end
+```
 
 Adding a list of items to our menu list is handled by a different function which takes in {self(), :additems, items}
```

Comment:
show each functionality individually and demostrate it here too to show the reader the complete story